reward/staking_info.go

```diff
@@ -44,21 +44,103 @@ var (
 
 // StakingInfo contains staking information.
 type StakingInfo struct {
-	BlockNum uint64 // Block number where staking information of Council is fetched
+	BlockNum uint64 `json:"blockNum"` // Block number where staking information of Council is fetched
 
 	// Information retrieved from AddressBook smart contract
-	CouncilNodeAddrs    []common.Address // NodeIds of Council
-	CouncilStakingAddrs []common.Address // Address of Staking account which holds staking balance
-	CouncilRewardAddrs  []common.Address // Address of Council account which will get block reward
-	// TODO-klaytn-fund: Provide backward-compatibility to read/write marshalled stakingInfo, and then change KIR/PoC to KCF/KFF
-	KIRAddr common.Address // Address of KIR contract
-	PoCAddr common.Address // Address of PoC contract
+	CouncilNodeAddrs    []common.Address `json:"councilNodeAddrs"`    // NodeIds of Council
+	CouncilStakingAddrs []common.Address `json:"councilStakingAddrs"` // Address of Staking account which holds staking balance
+	CouncilRewardAddrs  []common.Address `json:"councilRewardAddrs"`  // Address of Council account which will get block reward
 
-	UseGini bool
-	Gini    float64 // gini coefficient
+	KCFAddr common.Address `json:"kcfAddr"` // Address of KCF contract
+	KFFAddr common.Address `json:"kffAddr"` // Address of KFF contract
+
+	UseGini bool    `json:"useGini"`
+	Gini    float64 `json:"gini"` // gini coefficient
 
 	// Derived from CouncilStakingAddrs
-	CouncilStakingAmounts []uint64 // Staking amounts of Council
+	CouncilStakingAmounts []uint64 `json:"councilStakingAmounts"` // Staking amounts of Council
+}
+
+// MarshalJSON supports json marshalling for both oldStakingInfo and StakingInfo
+// TODO-klaytn-Mantle: remove this marshal function when backward-compatibility for KIR/PoC is not needed
+func (st StakingInfo) MarshalJSON() ([]byte, error) {
```

Comment:
```suggestion
func (st *StakingInfo) MarshalJSON() ([]byte, error) {
```

Reply:
MarshalJSON need to be a struct, not a pointer. Please see https://groups.google.com/g/golang-nuts/c/gW53LC_ZC98

Reply:
I see. Then, there are lots of MarshalJSON with pointers in the klaytn repo that need to get fixed. For the reference, here's to see that it doesn't work:

```go
package main

import (
	"encoding/json"
	"fmt"
	"reflect"
)

type A struct{ a int }

func (a *A) MarshalJSON() ([]byte, error) {
	fmt.Println(a.a)
	return json.Marshal(a.a)
}

func main() {
	{
		x := interface{}(A{a: 10})
		fmt.Println(x.(A).a) // This works
		//fmt.Println(&x.(A).a) // This doesn't

		t := reflect.TypeOf(x)
		fmt.Println("Num methods", t.NumMethod())

		b, err := json.Marshal(x)
		fmt.Println(string(b), err)
	}
	fmt.Println("==================")
	{
		x := A{a: 10}
		t := reflect.TypeOf(x)
		fmt.Println("Num methods", t.NumMethod())

		b, err := json.Marshal(&x)
		fmt.Println(string(b), err)
	}
	fmt.Println("==================")
	{
		x := &A{a: 10}
		t := reflect.TypeOf(x)
		fmt.Println("Num methods", t.NumMethod())

		for i := 0; i < t.NumMethod(); i++ {
			fmt.Println("...", t.Method(i).Name)
		}

		b, err := json.Marshal(&x)
		fmt.Println(string(b), err)
	}
}
```